File: Round2/B/B.cpp
#include<bits/stdc++.h>
using namespace std;

const int N = 310;

int n, m, a[N];
char s[110], ans[2][10] = {"Alice", "Bob"};

int solve(){
	int cnt = 0;
	for(int i=1; i<=m; i++){
		if(a[i] == 1)   return 0;
		if(a[i] == 2)   cnt++;
	}
	if(cnt == 0)    return 1;
	if(cnt == 1){
	if(strcmp(s, ans[0]) == 0)  return 0;
		return 1;
	}
	return 0;
}

int main(){
	scanf("%d %s", &n, s);

	memset(a, 0, sizeof(a));
	m = 0;
	int x = 0;
	for(int i=0; i<n; i++){
		scanf("%d", &x);
		m = max(m, x);
		a[x]++;
	}

	puts(ans[solve()]);

	return 0;
}

File: Round2/B/b.php
<?php

list($n, $name) = explode(' ', fgets(STDIN));
$n = intval($n);
$name = trim($name);

$arr = explode(' ', fgets(STDIN));
$A = [];
foreach($arr as $v){
	$v = intval($v);
	if(!isset($A[$v])){
		$A[$v] = 0;
	}
	$A[$v]++;
}

function solve($A, $name){
	$cnt = 0;
	foreach($A as $v){
		if($v == 1)	return 'Alice';
		if($v == 2)	$cnt++;
	}
	if($cnt == 0)	return 'Bob';
	if($cnt == 1){
		return $name;
	}
	return 'Alice';
}

printf("%s\n", solve($A, $name));